src/kvstore.c

```diff
@@ -729,11 +733,15 @@ unsigned int kvstoreHashtableSampleEntries(kvstore *kvs, int didx, void **dst, u
 }
 
 bool kvstoreHashtableExpand(kvstore *kvs, int didx, unsigned long size) {
-    hashtable *ht = kvstoreGetHashtable(kvs, didx);
-    if (!ht) return false;
+    hashtable *ht = createHashtableIfNeeded(kvs, didx);
     return hashtableExpand(ht, size);
 }
 
+bool kvstoreHashtableTryExpand(kvstore *kvs, int didx, unsigned long size) {
+    hashtable *ht = createHashtableIfNeeded(kvs, didx);
```

Comment:
Skip if `size == 0`?

```suggestion
    if (size == 0) return false;
    hashtable *ht = createHashtableIfNeeded(kvs, didx);
```

Reply:
Is this incorrect API usage altogether? Should we crash if the expand request is for `0` ?

Reply:
We can decide what's correct usage.

I think we can let this function accept 0 and return false. The API is simpler this way. You can call it with any size without exceptions.